File: Plantillas/regression_template.py
# -*- coding: utf-8 -*-
"""
Created on Wed Jun 22 18:00:40 2022

@author: Pelka
"""

# Import libraries

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importar dateset

dataset = pd.read_csv('Position_Salaries.csv')
x = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

# Dividir el data set en conjunto de entrenamiento y conjunto de testing

from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, random_state = 0)

# Escalado de variables
"""
from sklearn.preprocessing import StandardScaler
sc_x = StandardScaler()
x_train = sc_x.fit_transform(x_train)
x_test = sc_x.transform(x_test)"""

# Ajustar la regresion con el dataset


# Prediccion de nuestro modelo

y_pred = regression.predict()

# Visualizacion de los resultados del Modelo Polinomico

x_grid = np.arange(min(x), max(x), 0.1)
x_grid = x_grid.reshape(len(x_grid), 1)
plt.scatter(x, y, color = "red")
plt.plot(x_grid, regression.predict(), color ="blue")
plt.title("Modelo de Regresion")
plt.xlabel("lblx")
plt.ylabel("lbly")
plt.show()

File: Plantillas/regression_template.r
# Plantilla de Regresion

# Import dataset

dataset = read.csv('S4 - Polynomial Regression/Position_Salaries.csv')
dataset = dataset[, 2:3]

# Dividir los datos en conjunto de entrenamiento y conjunto de test
# 
# library(caTools)
# set.seed(123)
# split = sample.split(dataset$Salary, SplitRatio = 2/3)
# training_set = subset(dataset, split == TRUE)
# testing_set = subset(dataset, split == FALSE)

# Ajustar modelo de Regresion con el mismo conjunto de datos

# Crear nuestra variable de regresion




# Prediccion de los resultados --------------------------------------------

y_pred_lin = predict(lin_reg, newdata = data.frame(Level = 6.5))



# Visualizacion del modelo de regresion -----------------------------------

library(ggplot2)

x_grid = seq(min(dataset$value), max(dataset$values), 0.1)

ggplot() +
  geom_point(aes(x = dataset$Level , y = dataset$Salary), 
             color = "red") +
  geom_line(aes(x = x_grid , y = predict(regression, 
                                         newdata = data.frame(value = x_grid))),
            color = "blue") +
  ggtitle("Prediccion") +
  xlab("") +
  ylab("")




